core/shared/src/main/scala/fs2/async/RefOps.scala

```diff
@@ -0,0 +1,49 @@
+package fs2.async
+
+/**
+ * Operations on concurrent, mutable reference types.
+ */
+trait RefOps[F[_], A] {
```

Comment:
Note there's no `setAsync` operation, as that's not possible to implement on `SyncRef`